apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,57 @@
 'use strict';
 
 $(document).ready(function() {
-  // Put app logic here
+  var $block = null;
+
+  $('[data-stack]').click(function() {
+    //this is a listener for clicks!
+
+    if ($block == null) {
+      //setting the announcement text to nothing in case the player made an illegal move previously
+      $('#announce-game-won').text('');
+      $block = $(this).children().last().detach();
+      //'this' is the data stack that was clicked on and will return the jquery object that we need
+      //we are picking up a block
+    } else {
+      //we are putting the block back down
+      if (isLegalMove($block, $(this))) {
+        $(this).append($block);
+
+        //setting the $block back to null so we can continue to play the game.
+        $block = null;
+      } else {}
+    }
+
+    checkForWin();
+  });
+
+  function checkForWin() {
+    //checking length of data stacks 2 and 3. if they equal 4, then the player has won the game.
+    if (
+      $('[data-stack="2"]').children().length === 4 || $('[data-stack="3"]').children().length === 4
+    ) {
+      //update test to out You Won!
+      $('#announce-game-won').text('You Won!');
+    }
+
+  }
+
+  function isLegalMove($heldBlock, $targetStack) {
+
+    if ( //puting a block down and the stack is empty, do it
```

Comment:
I like the if/else if structure choice here. Many students chose to do a single if with a big OR operation. I like this better, it reads very nicely